crates/compute/src/alloc.rs

```diff
@@ -32,7 +32,7 @@ pub trait ComputeAllocator<F, Mem: ComputeMemory<F>> {
 /// Basic bump allocator that allocates slices from an underlying memory buffer provided at
 /// construction.
 pub struct BumpAllocator<'a, F, Mem: ComputeMemory<F>> {
-	buffer: Cell<Option<Mem::FSliceMut<'a>>>,
+	buffer: Mutex<Option<Mem::FSliceMut<'a>>>,
```

Comment:
@BenjaminTrapani @alsrdn FYI

Reply:
Would it be possible to partition the bump allocator between threads? Doing so would remove the need for `FSlice` and `ComputeLayer` to be `Sync`. 

I am working on a change now that removes `host_alloc` from the compute layer. Once this change is complete, all Binius logic operates on an implementation of `ComputeAllocator` directly. If `ComputeAllocator` provides a way of partitioning the remaining space, each thread can receive its own free block, removing the synchronization overhead in the multi-threaded implementation.

Reply:
I see that this change is primarily for device allocations. The same approach of partitioning the remaining space between threads applies to device allocations as well.